### DISCUSSION TITLE
Another way to select whole words with the mouse

### DISCUSSION THREAD

**Discussion by ntrel** [geany/geany#3693] [Show and tell]
I just discovered something. I knew you can double click on a word and drag to start selecting by whole words, and this works across many apps.

But on my Linux Mint 21.2 system, Geany seems to be unique in allowing the following:
* Single click and drag on a word character, hovering over another word character
* Then release and immediately click again without moving the mouse

Geany will extend the selection to include the whole of any partially selected words.

I expect the behaviour is part of Scintilla, anyone have more details, when it was added? I expect it isn't documented in Geany's manual.

This is useful when you start a normal selection and either:
* you can't be bothered to double click (I avoid it sometimes as you have to be more precise with your finger than a single click and ensure you don't move the mouse before the second click)
* after selecting most of the text you want, you want to switch to whole-word selection mode rather than keep dragging and have to be precise about where you release the mouse button.

So this gives another way to reduce the distance you have to move the mouse besides the initial double clicking method.

**Comment by elextr** [comment]
On LM it works on 1.38 as well.

Maybe LM default time between button release and a second click to to be treated as double click is set longer so its easier to do.

**Reply by ntrel** [reply]
I don't think it's just that because you can have multiple words selected, including extending the selection at the start to the left.

**Comment by elextr** [comment]
Geany does not set or cancel [selection mode](https://www.scintilla.org/ScintillaDoc.html#SCI_SETSELECTIONMODE) so maybe Scintilla is defaulting to 0 so extend selection is now available?
